daemon/src/wallet_seed.rs

```diff
@@ -0,0 +1,89 @@
+use anyhow::bail;
+use anyhow::Result;
+use bdk::bitcoin::util::bip32::ExtendedPrivKey;
+use bdk::bitcoin::Network;
+use bip39::Language;
+use bip39::Mnemonic;
+use rand::thread_rng;
+use std::path::Path;
+use std::str::FromStr;
+
+#[derive(Clone)]
+pub struct WalletSeed(Mnemonic);
```

Comment:
Something to mull over:

Wrapping a type is useful if you want maintain certain invariants because you can limit the functions that are called on it and put additional checks before and after.

In this case, we are

a) allowing to pass in any arbitrary `Mnemonic`
b) exposing the inner instance

and hence not enforcing any invariants. That could be a hint that we are wrapping this type unnecessarily. In such situations an alternative, flatter design would be to use an extension trait:

```rust
pub trait MnemonicExt {
	async fn initialize(seed_file: &Path) -> Self;
	/// etc
}

impl MnemonicExt for Mnemonic {

}
```

Reply:
The benefit of an extension trait is that there is no change in hierachy and a user can use other, already existing APIs without having to wrap / unwrap a variable.

Reply:
cheers, this a pretty cool pattern/idea